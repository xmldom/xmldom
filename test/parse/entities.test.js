'use strict';

const { getTestParser } = require('../get-test-parser');
const { MIME_TYPE } = require('../../lib/conventions');

describe('entity replacement ignores js prototype chain', () => {
	it("should not pick up 'entities' from the prototype chain", () => {
		const source = `
		<xml>
			<hasOwnProperty>&hasOwnProperty;</hasOwnProperty> 
			<proto>&__proto__;</proto> 
			<constructor>&constructor;</constructor>
		</xml>
`;
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString(source, MIME_TYPE.XML_TEXT).toString();

		expect({ actual, ...errors }).toMatchSnapshot();
	});
});
