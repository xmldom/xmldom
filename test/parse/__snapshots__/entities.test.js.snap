// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity replacement ignores js prototype chain should not pick up 'entities' from the prototype chain 1`] = `
Object {
  "actual": "
		<xml>
			<hasOwnProperty>&amp;hasOwnProperty;</hasOwnProperty> 
			<proto>&amp;__proto__;</proto> 
			<constructor>&amp;constructor;</constructor>
		</xml>",
  "errors": Array [
    Array [
      "error",
      "entity not found:&hasOwnProperty;",
      Object {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
    Array [
      "error",
      "entity not found:&__proto__;",
      Object {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
    Array [
      "error",
      "entity not found:&constructor;",
      Object {
        "columnNumber": 44,
        "lineNumber": 5,
      },
    ],
  ],
}
`;
